src/lib/mathlib/math/Functions.hpp

```diff
@@ -166,28 +166,47 @@ const T interpolate(const T &value, const T &x_low, const T &x_high, const T &y_
 }
 
 /*
- * Constant, linear, linear, constant function with the three corner points as parameters
- *  y_high               -------
- *                      /
- *                    /
- *  y_middle        /
- *                /
+ * Constant, piecewise linear, constant function with 1/N size intervalls and N corner points as parameters
+ * y[N]                 -------
+ *                     /
+ *                   /
+ * y[1]            /
  *               /
  *              /
- * y_low -------
- *         x_low x_middle x_high
+ *             /
+ * y[0] -------
+ *        0 1/(N-1) 2/(N-1) ... 1
  */
-template<typename T>
-const T interpolate3(const T &value,
-		     const T &x_low, const T &x_middle, const T &x_high,
-		     const T &y_low, const T &y_middle, const T &y_high)
+template<typename T, size_t N>
+const T interpolateN(const T &value, const T(&y)[N])
 {
-	if (value < x_middle) {
-		return interpolate(value, x_low, x_middle, y_low, y_middle);
+	size_t index = constrain((int)(value * (N - 1)), 0, (int)(N - 2));
+	return interpolate(value, (T)index / (T)(N - 1), (T)(index + 1) / (T)(N - 1), y[index], y[index + 1]);
```

Comment:
```suggestion
	size_t index = constrain((int)(value * N), 0, (int)(N - 1));
	return interpolate(value, (T)index / (T)(N), (T)(index + 1) / (T)(N), y[index], y[index + 1]);
```

Why is the divider `N - 1`? Isn't the range divided into N different sections? E.g. for Y and X each:
- Y[0] ... Y[N]
- 0 ... (N/N)

Reply:
Ah, that is necessary because the index can not reach [N]!

Then the `y[N]` should be named `y[N-1]` in the comment